go/service/assets/api_op_CreateAsset.go

```diff
@@ -107,6 +107,10 @@ func (c *Client) CreateAsset(createAssetInput CreateAssetInput) (*CreateAssetOut
 		return nil, fmt.Errorf("error while reading response body: %v", err)
 	}
 
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("ERROR: got %d from %s %s endpoint . Full response : \n %s", res.StatusCode, res.Request.Method, endpoint, response)
+	}
```

Comment:
Since `http.StatusOK` is equal to `200`, having a status code equal to `201` for example, should not raise an exception, I assume. That's why I suggest maybe doing it in this way.
```suggestion
statusOk := res.StatusCode == 200 && res.StatusCode < 300
	if !statusOk {
		return nil, fmt.Errorf("error: got %d from %s %s endpoint . Full response : \n %s", res.StatusCode, res.Request.Method, endpoint, response)
	}
```
Also, another note, I think error messages should start in lowercase in general. wdyt?

Reply:
I see that starting with lowercase is golang convention so I will adjust that accordingly.

However, with regards to status codes, I believe in holding the API to strict(er) contracts. This ensures we are bound to the same interface, and has a side-effect of simplifying (even slightly) our code. 

For your example, while 200 and 201 both signal success, they vary in the detail that a 201 means (at least in principle) that something was created. This behavior means that the request has a side-effect and we should be aware of it.
curious to hear your thoughts on this.

Reply:
Yes, fair enough we can go with that. 👌